File: lectures/Clase 03 (07-09-2018) - Verificación de programas secuenciales/progs_clase_03/divmod_v0.c
#include <stdio.h>

void divmod(int x, int y, int* q, int* r)
{
    *r=x; *q=0;
    while (*r>y) {
        *r=*r-y; (*q)++;
    }
}

void main(void)
{
    int x, y=7, q, r, i;

    for (i=0; i<5; i++) {
        x=100+i;
        divmod(x,y,&q,&r); 
        printf("%d = %d*%d + %d\n",x,y,q,r);
    }
}

File: lectures/Clase 03 (07-09-2018) - Verificación de programas secuenciales/progs_clase_03/divmod_v0.py
def divMod(x,y):
    r,q= x,0
    while r>y:
        r,q= r-y,q+1
    return q,r

q,r= divmod(22,7)
print()
print("divmod (builtin function): 22 = 7 *",q,'+',r)
print()

for i in range(5):
    q,r= divMod(100+i,7)
    print("%d = %d*%d + %d" % (100+i,7,q,r))
